src/Schema/Directives/Nodes/UnionDirective.php

```diff
@@ -46,10 +42,26 @@ function() use ($value){
                     return resolve(TypeRegistry::class)->get($type->name->value);
                 })->filter()->toArray();
             },
+            'resolveType' => $this->getResolver(function ($value, Context $context, ResolveInfo $info) {
+                $unionName = $this->definitionNode->name->value;
 
-            'resolveType' => function ($value) use ($resolver) {
-                return $resolver($value);
-            },
+                // Try to locate a fallback resolver corresponds to the config file.
+                $resolverClass = config('lighthouse.namespaces.unions') . '\\' . $unionName;
+
+                if (\class_exists($resolverClass)) {
+                    $resolver = resolve($resolverClass);
+
+                    if ( ! \method_exists($resolver, 'resolve')) {
```

Comment:
Call this with the class name before resolving the class, it will avoid unnecessary work.